pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2393,6 +2396,12 @@ type KafkaEndpointSpec struct {
 	// +kubebuilder:default=PLAIN
 	// +optional
 	Mechanism string `json:"mechanism,omitempty"`
+	// The kafka user name to use for authentication
+	// +optional
+	UserRef *corev1.SecretKeySelector `json:"UserRef,omitempty"`
```

Comment:
I struggled with what the field names should be. I believe that a SecretKeySelector field is most often named `SecretRef` but we need to add another struct here to follow that convention.  Note that the the attributes names for these fields added in Squid are `user-name` and `password`.  Should these be changed to something like `UsernameRef` and `PasswordRef`?

Reply:
`UserSecretRef`, `PasswordSecretRef`?

Reply:
I would also vote for `UserSecretRef`, `PasswordSecretRef`

Reply:
Done.